### PR TITLE
AArch64: Setup Travis CI Cross Compile Build

### PR DESCRIPTION
**Pull request by aarongraham9**
Add in Travis CI Build Cross-Compile for AArch64

Part of: #2446

Signed-off-by: Aaron Graham <aaron.graham@unb.ca>

### CODE REVIEW

File: include_core/AtomicSupport.hpp [hunk 1]
```diff
@@ -312,7 +312,7 @@ class VM_AtomicSupport
 #elif defined(ARM) /* defined(J9HAMMER) */
 		__sync_synchronize();
 #elif defined(AARCH64) /* defined(ARM) */
-		__asm __volatile ("dmb ishld":::"memory")
+		__asm __volatile ("dmb ishld":::"memory");
```
Comment (rwy0717) [inline include_core/AtomicSupport.hpp]: I think these^^ changes belong in a separate commit, it's totally unrelated to CI.

**Comment by rwy0717** [comment]
build-on-travis is starting to look pretty hairy. It might be time to break the script up into a few functions.

Also, I noticed that you went with an autotools build, and not cmake. Any reason for that choice? Maybe we want both (@charliegracie)?

File: scripts/build-on-travis.sh [hunk 2]
```diff
@@ -56,6 +56,67 @@ else
...
+    wget http://ftp.us.debian.org/debian/pool/main/libx/libxrender/libxrender-dev_0.9.10-1_arm64.deb
+    wget http://ftp.us.debian.org/debian/pool/main/libx/libxt/libxt-dev_1.1.5-1_arm64.deb
+    wget http://ftp.us.debian.org/debian/pool/main/libx/libxtst/libxtst6_1.2.3-1_arm64.deb
+    wget http://ftp.us.debian.org/debian/pool/main/libx/libxtst/libxtst-dev_1.2.3-1_arm64.deb
+    wget http://ftp.us.debian.org/debian/pool/main/x/x11proto-core/x11proto-core-dev_7.0.31-1_all.deb
+    wget http://ftp.us.debian.org/debian/pool/main/x/x11proto-input/x11proto-input-dev_2.3.2-1_all.deb
+    wget http://ftp.us.debian.org/debian/pool/main/x/x11proto-kb/x11proto-kb-dev_1.0.7-1_all.deb
+    wget http://ftp.us.debian.org/debian/pool/main/x/x11proto-render/x11proto-render-dev_0.11.1-2_all.deb
+    wget http://ftp.us.debian.org/debian/pool/main/x/x11proto-xext/x11proto-xext-dev_7.3.0-1_all.deb
+    wget http://ftp.us.debian.org/debian/pool/main/z/zlib/zlib1g_1.2.8.dfsg-5_arm64.deb
```
[the remark this answers is not in the document]
Comment (JamesKingdon) [reply scripts/build-on-travis.sh]: Those are probably down to me, back when I was getting the first armhf cross-compile working for OpenJ9. I think they were all required for OpenJ9, but it seems quite likely that only a subset are needed for OMR.

[the remark this answers is not in the document]
Comment (aarongraham9) [reply scripts/build-on-travis.sh]: Yes. The Cross-Compile script derives from @JamesKingdon's Dockerfile (https://github.com/eclipse/openj9/blob/2ff002784a0dc1f3439c62972637b2833dc76956/buildenv/docker/jdk9/aarch64_CC/arm-linux-aarch64/Dockerfile) in the OpenJ9 repo. I agree most  of these dependencies are required for OpenJ9 but not OMR. I'll try and determine which we require and cleanup the remainder.

File: include_core/AtomicSupport.hpp (hunk 1 above)
Comment (aarongraham9) [reply include_core/AtomicSupport.hpp]: Ok. I'll split them out in another PR. They were inadvertently missed in the Build and/or Part PR(s).

**Comment by aarongraham9** [comment]
Agreed @rwy0717 I can certainly split it out in a function. No reason in particular for autotools build. Just that the cross-compiler builds and scripts that @jeanlego and I are currently doing/using are with the autotools method. We are certainly open to either switch it over or use both; It's up to you, @0xdaryl and @charliegracie.

File: scripts/build-on-travis.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (rwy0717) [reply scripts/build-on-travis.sh]: Sounds good. At a glance, I don't think we need even a single one of these libraries.

File: include_core/AtomicSupport.hpp (hunk 1 above)
Comment (aarongraham9) [reply include_core/AtomicSupport.hpp]: This was moved to and merged in PR #2549.

**Comment by aarongraham9** [comment]
Thanks @rwy0717!

**Comment by aarongraham9** [comment]
The build passed with the amended code; would some of you be able to look at this for a review and/or a possible merge @charliegracie, @youngar, @0xdaryl, @JamesKingdon and @jeanlego?

**Comment by JamesKingdon** [comment]
Wow, that sure looks a lot simpler.

**Comment by aarongraham9** [comment]
Indeed it is!

**Comment by 0xdaryl** [comment]
Please squash the commits.

**Comment by aarongraham9** [comment]
Copyright updated. Commits Squashed. Should be good to go pending Appveyor and Travis builds.

**Comment by aarongraham9** [comment]
Build passed. Any more thoughts? @charliegracie would you be able to take a quick look; I believe you modified these files last.

**Comment by aarongraham9** [comment]
Thanks!

**Pull request merged**